File: 3083-existence-of-a-substring-in-a-string-and-its-reverse/3083-existence-of-a-substring-in-a-string-and-its-reverse.cpp
class Solution {
public:
    bool isSubstringPresent(string s) {
        int n=s.size(),i;
        set<string> st;
        for(i=0;i<n-1;i++){
            string x=s.substr(i,2);
            st.insert(x);
        }
        for(i=n-1;i>0;i--){
            string x;
            x+=s[i];
            x+=s[i-1];
            if(st.count(x)){
                return 1;
            }
        }
        return 0;
    }
};

File: 3083-existence-of-a-substring-in-a-string-and-its-reverse/3083-existence-of-a-substring-in-a-string-and-its-reverse.java
class Solution {
    public boolean isSubstringPresent(String s) {
        int n=s.length();
        int i=0;
        
        HashSet<String> st = new HashSet<>(); 
        for(i=1;i<n;i++){
            // String s1="";
            StringBuffer s1 = new StringBuffer();
            s1.append(s.charAt(i-1));
            // s1=s1+s[i];
            s1.append(s.charAt(i));
            st.add(s1.toString());
        }
        for(i=n-1;i>0;i--){
            // String s1="";
            StringBuffer s1 = new StringBuffer();
            // s1+=s[i];
            s1.append(s.charAt(i));
            // s1+=s[i-1];
            s1.append(s.charAt(i-1));
            if(st.contains(s1.toString())){
                return true;
            }
        }
        return false;
    }
}